Dashboard Page

Ewu694 opened the pull request:
Dashboard page completed:
- Nav component containing tabs linking to our other features like the chatbot and flashcards
- Tab to take user back to the home page '/'
- Tab to take users to analytics (To be added in future)
- Added API call to fetch user submissions from api.algoarchive.org
- Made cards to display all of a user's submissions within a repo
- Pagination implemented for larger amounts of submissions
- Searchbar design added, functionality is yet to be included (To be determined)

![image](https://github.com/user-attachments/assets/5ad61ae2-8edc-44ad-b875-ef9fcdf9cfbd)
![image](https://github.com/user-attachments/assets/efcf46ed-6751-4286-a67a-54bc8bab380f)

Ewu694 committed:
Dashboard Page Init

Dashboard Navbar basic functionality completed
- Home/Logo tab added to take users back home on click
- Added dropdown to Features tab to let users use other features of AlgoArchive

Other:
- Installed NextUI
- Added Icon.tsx to store icons that we may use throughout the development process
- Added our logo

Ewu694 committed:
Update tailwind.config.ts

Changed BG-color of Navbar to make Logo and other navbar items pop more

Ewu694 committed:
Changed from navbar to sidebar

Same functionality but now its a sidebar!!

Ewu694 committed:
Navbar upgrades

Added analytics tab and added another line above where the future Sign-Out button will be to better organize navbar visually

Ewu694 committed:
Reverted Changes back to regular Navbar

Reverted from sidenav to previous nav

Added in Analytics tab to previous nav

Ewu694 committed:
Hero Section Init

Hero Section for generating submissions initialized

Ewu694 committed:
User submissions FE integration

User Submissions API initialized into FE
Bonus: Search Bar more interactive

Ewu694 committed:
Create route.ts

Successfully integrated API call from api.algoarchive.org to pull user solutions

Ewu694 committed:
Integrated user submissions display

Integrated Submissions to display within the Hero.tsx
- Submissions displayed in cards
TBD: Card content will be updated after @riimuru
Styling could use some work too...

Ewu694 committed:
Feature: Pagination for submissions

- Implemented NextUI's pagination component
- Implemented Paginaiton Logic in PaginationControls.tsx using useSearchParams from NextJS
- Made it so the submissions component displays "fetching submissions" until all of the user's submissions are fetched and ready to be displayed

Ewu694 committed:
Update Hero.tsx

Made userName bolder to make it stick out more

Ewu694 committed:
Update Submissions.tsx

Stylization Updates for cards

riimuru reviewed (dismissed):
Great job on the design! Can you run `next lint` or `next lint --fix` to resolve any lining errors and ensure Cloudflare Pages can run smoothly?

Ewu694 committed:
Merge branch 'main' into dashboard

Ewu694 committed:
Updated json to take in eslint plugin for next

Fixed linting errors and installed ESLint/Next Lint

Ewu694 committed:
ESLinting Fixed

Updated Packages and Centered "Failed Fetching Submissions" Message

Ewu694 committed:
Update package-lock.json

NPM CI and NPM Updated

Ewu694 committed:
ESLint Fix part 2

Removed Typescript Parser since CloudFare does not support this version with their node version

riimuru committed:
chore: npm packages

Ewu694 committed:
Typescript Type Errors Fixed?

Fixed Typescript and Linting errors for 'any' types

Ewu694 committed:
Merge branch 'dashboard' of https://github.com/AlgoArchiveExt/algoarchive.org into dashboard

Ewu694 committed:
Update Hero.tsx

Wrapped Submissions which uses useSearchParams in Suspense

Ewu694 committed:
Prettier Linting Error Fixed

Ewu694 committed:
Update Hero.tsx

Ewu694 committed:
Update route.ts

Fixed Edge error in route.ts for Cloudfare Deployment

Ewu694 committed:
Update route.ts

Another Cloudfare error fix

Ewu694 committed:
Update route.ts

Changed Handler to GET, did an oopsie

Ewu694 committed:
Updated Route

- Updated Route from localhost to deployed AWS API server
- Fixed styling error for Pagination controls

Ewu694 committed:
Merge branch 'main' into dashboard

Ewu694 committed:
Merge Conflict Duplication Errors Fixed

Fixed Merge Conflict error duplication bugs

Ewu694 committed:
Fixed Linting errors from PR#11

- Prettier ESLint problems resolved
- Class and Naming errors resolved

Ewu694 committed:
Update package-lock.json

Chore: npm packages
